File: mobile/lib/wasteOrder/Service.dart
class Service {
  final String description;
  final String description2;
  final String materialDescription;
  final int equipmentType;
  final String equipmentDescription;

  Service(
      {this.description,
      this.description2,
      this.materialDescription,
      this.equipmentType,
      this.equipmentDescription});

  factory Service.fromJSON(Map<String, dynamic> json) {
    return Service(
        description: json['description'],
        description2: json['description2'],
        materialDescription: json['materialDescription'],
        equipmentType: json['equipmentType'],
        equipmentDescription: json['equipmentDescription']);
  }
}

File: application/src/wasteOrder/Service.ts
import * as Joi from '@hapi/joi';

export enum EquipmentType {
    SUBMISSION,
    PICK_UP,
    EXCHANGE,
    CLEARANCE
}

export interface Service {
    description: string;
    description2: string;
    materialDescription: string;
    equipmentType: EquipmentType;
    equipmentDescription: string;
}

/**
 *  @swagger
 *  definitions:
 *      ServiceSchema:
 *          type: object
 *          required:
 *              - description
 *              - description2
 *              - materialDescription
 *              - equipmentType
 *              - equipmentDescription
 *          properties:
 *              description:
 *                  type: string
 *              description2:
 *                  type: string
 *              materialDescription:
 *                  type: string
 *              equipmentType:
 *                  type: integer
 *              equipmentDescription:
 *                  type: string
 */
export const ServiceSchema = Joi.object().keys({
    description: Joi.string().required(),
    description2: Joi.string().required(),
    materialDescription: Joi.string().required(),
    equipmentType: Joi.number().required(),
    equipmentDescription: Joi.string().required()
});